Assets/MixedRealityToolkit-SDK/Features/UX/Scripts/Controllers/MixedRealityControllerVisualizer.cs

```diff
@@ -0,0 +1,285 @@
+using Microsoft.MixedReality.Toolkit.Internal.Interfaces;
+using Microsoft.MixedReality.Toolkit.Internal.Interfaces.InputSystem.Handlers;
+using Microsoft.MixedReality.Toolkit.Internal.Managers;
+using Microsoft.MixedReality.Toolkit.Internal.Utilities;
+using Microsoft.MixedReality.Toolkit.SDK.Input;
+using UnityEditor;
+using UnityEngine;
+
+namespace Microsoft.MixedReality.Toolkit.SDK.UX.Controllers
+{
+    /// <summary>
+    /// The Mixed Reality Visualization component is primarily responsible for rendering the users controllers in a scene.
+    /// It will ensure whether you have a specific model configured against the controller definition and fall back to using the generic default models defined in the Controllers Profile.
+    /// The visualizer can be enabled / disabled on demand (after startup) 
+    /// </summary>
+    /// <example>
+    /// To Start, ensure there is a "Controllers Profile" configured against the main Mixed Reality Configuration Profile and that Controller Rendering is enabled in that profile, as well as configuring at least one controller for each platform you intend to support.
+    /// Once ready, then simply add this script to a GameObject in the scene, for example the Camera / head.
+    /// </example>
+    /// <remarks>
+    /// For Alpha, this visualizer only does basic rendering and management of controllers.  In future versions, it will expose specific transforms on a controller model (if it has them) to be able to attach items to any part of the model
+    /// It will also support animation of controller parts and possibly fading in and our of controllers.
+    /// </remarks>
+    /// <seealso cref="MixedRealityControllerMappingProfile"/>
+    public class MixedRealityControllerVisualizer : InputSystemGlobalListener, IMixedRealitySourcePoseHandler, IMixedRealityInputHandler
+    {
+
+        #region Private Properties
+
+        private static GameObject leftControllerModel;
+        private static IMixedRealityController leftController;
+        private static MixedRealityPose leftControllerOffsetPose = MixedRealityPose.ZeroIdentity;
+        private static GameObject rightControllerModel;
+        private static IMixedRealityController rightController;
+        private static MixedRealityPose rightControllerOffsetPose = MixedRealityPose.ZeroIdentity;
+
+        #endregion Private Properties
+
+        #region SourcePose Event Handlers
+
+        /// <summary>
+        /// The selected controller is moving in the scene, update it's pose in relation to the users movements
+        /// </summary>
+        /// <param name="eventData"></param>
+        public void OnSourcePoseChanged(SourcePoseEventData eventData)
+        {
+            //Handles.PositionHandle(eventData.MixedRealityPose.Position, eventData.MixedRealityPose.Rotation);
+
+            // Update the respective controller if it has been initialized
+            switch (eventData.Controller.ControllerHandedness)
```

Comment:
Wouldn't this be a bit cleaner in an if-else?

Reply:
Or do you plan on handling other handedness types?

Reply:
there needs, at least to be a default in this switch (unhanded, etc).

Reply:
Either it's handled or not.   Nothing happens if it's not handled, so no default is really required.


And yes, leaving the scope open for the visualizer to be enhanced later to support different types.